crates/hotshot/types/src/stake_table.rs

```diff
@@ -40,4 +47,123 @@ impl<K: SignatureKey> StakeTableEntry<K> {
     }
 }
 
-// TODO(Chengyu): add stake table snapshot here
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Default)]
+pub struct HSStakeTable<TYPES: NodeType>(pub Vec<PeerConfig<TYPES>>);
+
+impl<TYPES: NodeType> From<Vec<PeerConfig<TYPES>>> for HSStakeTable<TYPES> {
+    fn from(peers: Vec<PeerConfig<TYPES>>) -> Self {
+        Self(peers)
+    }
+}
+
+impl<TYPES: NodeType> std::ops::Deref for HSStakeTable<TYPES> {
```

Comment:
do you want to use `derive_more::Deref` to auto generate these: https://docs.rs/derive_more/latest/derive_more/derive.Deref.html

Reply:
done in [ae5454d](https://github.com/EspressoSystems/espresso-network/pull/3233/commits/ae5454dcb5ba9c67dc7aeda32459284df531c5d4)